iceoryx_utils/platform/mac/source/semaphore.cpp

```diff
@@ -289,7 +289,7 @@ int iox_sem_init(iox_sem_t* sem, int, unsigned int value)
     pthread_mutexattr_destroy(&mutexAttr);
 
     sem->m_hasPosixHandle = false;
-    sem->m_value.store(value, std::memory_order_relaxed);
+    sem->m_value.store(static_cast<int>(value), std::memory_order_relaxed);
```

Comment:
@elfenpiff is it on purpose that m_value has the fundamental type int and not int32_t?

Reply:
I guess the internal representation should match the type of the POSIX function which returns this.